content/articles/mocking-with-junit-and-mockito-the-why-and-how/index.md

```diff
@@ -0,0 +1,1011 @@
+The ```verifyNoInteractions(Object... mocks)``` method verifies that no method was invoked on the specified mock objects.
+The following snippet demonstrates using ```verifyNoInteractions``` with mock objects.
+
+```java
+studentService.findStudentById("MAT100419");
+verifyNoInteractions(hostelRepository);
+```
+#### Verifying the order of interactions
+Mockito provides an interface - [InOrder](https://www.javadoc.io/static/org.mockito/mockito-core/2.6.9/org/mockito/InOrder.html#verifyNoMoreInteractions())
+that allows us to verify the order of interactions with mock objects.
+
+There are three versions of ```Inorder```:
+* ```verify (Syntax: <T> T verify(T mock))```: This verifies that an interaction with a mock object happened once in order.
+  The following snippet demonstrates how to use the ```InOrder.verify``` method:
+
+```java
+void testTheOrderOfInteractions() throws Exception {
+  RegistrationRequest registrationRequest = new RegistrationRequest(
+          "John",
+          "Doe",
+          "securedPassword",
+          "MAT100419",
+          Gender.MALE);
+  StudentDto studentDto = studentService.registerStudent(registrationRequest);
+  studentService.assignBedSpace(studentDto);
+  InOrder inorder = inOrder(studentRepository, hostelRepository);
+  inorder.verify(studentRepository).findById(anyString());
+  inorder.verify(hostelRepository).returnAvailableMaleSpace();
+}
+```
+Here we are verifying that when assigning a bed space to a student, the ```studentRepository``` object is interacted with
+once by invoking the ```findById``` method with any string passed in as a parameter.
+Next, we are verifying that the ```hostelRepository``` object is interacted with exactly once by invoking the
+```returnAvailableMaleSpace``` method.
+
+* ```Verify``` with [verification mode](https://www.javadoc.io/static/org.mockito/mockito-core/2.6.9/org/mockito/verification/VerificationMode.html)
+  passed in as an argument ("`Syntax: T mock, VerificationMode mode` "): This verifies that interaction occurred with a mock object exactly once but with verification mode passed in as an argument
+
+```java
+@Test
+void testTheOrderOfInteractionWithVerificationModePassedAsAnArgument() throws Exception {
+  RegistrationRequest registrationRequest = new RegistrationRequest(
+          "John",
+          "Doe",
+          "securedPassword",
+          "MAT100419",
+          Gender.MALE);
+  StudentDto studentDto = studentService.registerStudent(registrationRequest);
+  studentService.assignBedSpace(studentDto);
+  InOrder inorder = inOrder(studentRepository, hostelRepository);
+  inorder.verify(studentRepository, times(1)).findById(anyString());
+  inorder.verify(hostelRepository, times(1)).returnAvailableMaleSpace();
+  inorder.verify(hostelRepository, never()).findHostelByName(“HALL 3”);
+}
+```
+
+Here we are verifying that  in other to assign a bed space to a student, first, the ```findById``` method is invoked on the mock
```

Comment:
Here it seems you used "other" other than using "order" i.e. "Here we are verifying that in order to assign..."